pkg/controller/plan/adapter/vsphere/builder.go

```diff
@@ -1574,9 +1576,22 @@ func (r *Builder) mergeSecrets(migrationSecret, migrationSecretNS, storageVendor
 			dst.Data["GOVMOMI_INSECURE"] = value
 		}
 	}
+
 	// Update secret1 with the merged data.
-	if err := r.Destination.Update(context.Background(), dst); err != nil {
-		return fmt.Errorf("failed to update secret1: %w", err)
+	maxRetries := 10
+	for i := 0; i <= maxRetries; i++ {
+		current := core.Secret{}
+		if err := r.Destination.Get(context.Background(), client.ObjectKey{
+			Name:      migrationSecret,
+			Namespace: migrationSecretNS}, &current); err != nil {
+			return fmt.Errorf("failed to get migration secret: %w", err)
+		}
+		current.Data = dst.Data
+		if err := r.Destination.Update(context.Background(), &current); err != nil {
+			r.Log.Info(fmt.Sprintf("Try numer %d - failed to update secret1: %v",i, err))
+			time.Sleep(2 * time.Second)
```

Comment:
The controller is running in a single process without go routines this will halt all migrations happening at that time. Trying to avoid the sleep as much as possible